x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/queries/SemanticQueryBuilderTests.java

```diff
@@ -0,0 +1,352 @@
+                            ]
+                        }
+                    }
+                }
+            """,
+        SEMANTIC_TEXT_FIELD,
+        TEXT_EMBEDDING_DIMENSION_COUNT,
+        generateTextEmbeddingVector(TEXT_EMBEDDING_DIMENSION_COUNT)
+    );
+
+    private static InferenceResultType inferenceResultType;
+
+    private enum InferenceResultType {
+        NONE,
+        SPARSE_EMBEDDING,
+        TEXT_EMBEDDING
+    }
+
+    private Integer queryTokenCount;
+
+    @BeforeClass
+    public static void setInferenceResultType() {
+        // The inference result type is a class variable because it is used when initializing additional mappings,
+        // which happens once per test suite run in AbstractBuilderTestCase#beforeTest as part of service holder creation.
+        inferenceResultType = randomFrom(InferenceResultType.values());
+    }
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        queryTokenCount = null;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> getPlugins() {
+        return List.of(InferencePlugin.class);
+    }
+
+    @Override
+    protected Settings createTestIndexSettings() {
+        // Randomize index version within compatible range
+        // we have to prefer CURRENT since with the range of versions we support it's rather unlikely to get the current actually.
+        IndexVersion indexVersionCreated = randomBoolean()
+            ? IndexVersion.current()
+            : IndexVersionUtils.randomVersionBetween(random(), NEW_SPARSE_VECTOR, IndexVersion.current());
+        return Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, indexVersionCreated).build();
+    }
+
+    @Override
+    protected void initializeAdditionalMappings(MapperService mapperService) throws IOException {
+        mapperService.merge(
+            "_doc",
+            new CompressedXContent(
+                Strings.toString(PutMappingRequest.simpleMapping(SEMANTIC_TEXT_FIELD, "type=semantic_text,inference_id=test_service"))
+            ),
+            MapperService.MergeReason.MAPPING_UPDATE
+        );
+
+        applyRandomInferenceResults(mapperService);
+    }
+
+    private void applyRandomInferenceResults(MapperService mapperService) {
+        // Parse random inference results (or no inference results) to set up the dynamic inference result mappings under the semantic text
+        // field
+        String source = switch (inferenceResultType) {
+            case SPARSE_EMBEDDING -> SPARSE_EMBEDDING_INFERENCE_RESULTS;
+            case TEXT_EMBEDDING -> TEXT_EMBEDDING_INFERENCE_RESULTS;
+            default -> null;
+        };
+
+        if (source != null) {
+            SourceToParse sourceToParse = new SourceToParse("test", new BytesArray(source), XContentType.JSON);
+            ParsedDocument parsedDocument = mapperService.documentMapper().parse(sourceToParse);
+            mapperService.merge(
```

Comment:
@jimczi The approach I was using previously to initialize the semantic text field's sub-mappings (in particular, the model settings) fails in a strange way now. After the mapping update here is applied, the mapping lookup contains a field mapper for `semantic.inference.inference.chunks.embeddings`, when I expect a field mapper for `semantic.inference.chunks.embeddings`. I think something in the mapping merge code is causing this, but I can't find anything obvious. Any ideas?

Reply:
Good catch, I opened https://github.com/elastic/elasticsearch/pull/107507 to fix the redundant path.